src/Symfony/Component/Serializer/Instantiator/Instantiator.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Serializer\Instantiator;
+
+use Symfony\Component\PropertyInfo\PropertyListExtractorInterface;
+use Symfony\Component\Serializer\Exception\MissingConstructorArgumentsException;
+use Symfony\Component\Serializer\Exception\RuntimeException;
+use Symfony\Component\Serializer\NameConverter\NameConverterInterface;
+use Symfony\Component\Serializer\Normalizer\AbstractObjectNormalizer;
+use Symfony\Component\Serializer\Normalizer\DenormalizerAwareInterface;
+use Symfony\Component\Serializer\Normalizer\DenormalizerAwareTrait;
+use Symfony\Component\Serializer\Normalizer\ObjectToPopulateTrait;
+
+/**
+ * @author Jérôme Desjardins <jewome62@gmail.com>
+ */
+class Instantiator implements InstantiatorInterface, DenormalizerAwareInterface
+{
+    public const DEFAULT_CONSTRUCTOR_ARGUMENTS = AbstractObjectNormalizer::DEFAULT_CONSTRUCTOR_ARGUMENTS;
+
+    use ObjectToPopulateTrait;
+    use DenormalizerAwareTrait;
+
+    private $classDiscriminatorResolver;
+    private $propertyListExtractor;
+    private $nameConverter;
+
+    public function __construct(PropertyListExtractorInterface $propertyListExtractor = null, NameConverterInterface $nameConverter = null)
+    {
+        $this->propertyListExtractor = $propertyListExtractor;
+        $this->nameConverter = $nameConverter;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function instantiate(string $class, $data, $format = null, array $context = [])
+    {
+        $allowedAttributes = $this->propertyListExtractor ? $this->propertyListExtractor->getProperties($class, $context) : null;
+        $reflectionClass = new \ReflectionClass($class);
+        $constructor = $reflectionClass->getConstructor();
+
+        if (null === $constructor) {
```

Comment:
You are not checking if constructor is public or not, in case it's not, you want to treat the object as if it hasn't a constructor, and use `(new \ReflectionClass($class))->newInstanceWithoutConstructor();` to spawn the object to populate. That's one of the most annoying recurring bug I encounter on a daily basis with the normalizer.